website/docs/components/app-side-nav/partials/code/how-to-use.md

```diff
@@ -0,0 +1,387 @@
+
+Each one of these states has CSS class names associated, and they’re used by the component code to determine what layout to render for the sidebar navigation.
+
+#### Content transitions between states
+
+The App Side Nav component **automatically**:
+
+- fades in/out the “actions” block in the header and the content injected in the body via “portals”.
+- swaps the toggle button icon from “menu” to “close” and moves it from one position to another
+
+Any other content in the App Side Nav needs to be **explicitly handled** by the consumers (in this way they have full control of the content they add, and they can customize the transition as they want/need).
+
+One possible way to do it is to use the **`hds-app-side-nav-hide-when-minimized` class**. This is a special class that can be applied to a DOM element so that it **automatically** fades in/out when the App Side Nav changes its “minimization” state.
+
+More specifically:
+
+- `minimized → maximized` transition: the content appears with a fade-in effect, when the width animation is already completed (the width is maximised)
+- `maximized → minimized` transition: the content disappears at once with no transition, before the width animation starts
+
+Another option is to use the **`isMinimized` parameter**, which is useful in those cases where the content is so custom/specialized that can’t just be faded in/out but needs to have a different kind of transition (eg. remain visible but change layout or respond to the width of the container). This value is passed down by the `<:header/body/footer>` named blocks as parameters, and can be used to build custom logic on the consumers’ side.
+
+See the code snippet for the `<:footer>` above for examples of both approaches.
+
+#### Advanced customization
+
+Some aspects of the responsiveness/animation/transition of the App Side Nav are parameterized in code via CSS custom properties. It means that _in theory_ they could be customized/overwritten. This though is **something that we don’t recommend**.
+
+Navigation is a cardinal element of the UI of an application, and its standardization across the different HashiCorp products is an important end goal. Also, the implementation of this component has required considerable efforts and investments, and we should preserve it as much as possible as a unifying element across products.
+
+If you find yourself in the situation of wanting/needing to customize or change the aspect or the behavior of the App Side Nav component, [contact the Design Systems Team](/about/support) for support.
+
+### Accessibility
+
+#### `aria` attributes
+
+The App Side Nav component already provides some of the required `aria` attributes. But other attributes are needed when declaring its content. Please refer to the [Accessibility](/components/app-side-nav?tab=accessibility) section for more details.
```

Comment:
Nit: There isn't really anything in the a11y tab describing what attributes are needed when declaring the content. Can you specify here?